app/src/main/java/nerd/tuxmobil/fahrplan/congress/models/Session.java

```diff
@@ -313,16 +337,22 @@ public String getLanguageContentDescription(@NonNull Context context) {
         if (TextUtils.isEmpty(lang)) {
             return context.getString(R.string.session_list_item_language_unknown_content_description);
         }
+        String languageName = lang;
         if ("en".equals(lang)) {
-            return context.getString(R.string.session_list_item_language_english_content_description);
+            languageName = context.getString(R.string.session_list_item_language_english_content_description);
         }
         if ("de".equals(lang)) {
-            return context.getString(R.string.session_list_item_language_german_content_description);
+            languageName = context.getString(R.string.session_list_item_language_german_content_description);
         }
         if ("pt".equals(lang)) {
-            return context.getString(R.string.session_list_item_language_portuguese_content_description);
+            languageName = context.getString(R.string.session_list_item_language_portuguese_content_description);
         }
```

Comment:
Just a thought -> the if branches should be exclusive, right...maybe if-ifelse would has sense, also if this file would be converted to Kotlin it would be better for the Converter, since it would know to use `when` here instead.

Reply:
I can use a `switch` here instead:

``` java
String languageName = lang;
switch (lang) {
    case "en":
        languageName = context.getString(R.string.session_list_item_language_english_content_description);
        break;
    case "de":
        languageName = context.getString(R.string.session_list_item_language_german_content_description);
        break;
    case "pt":
        languageName = context.getString(R.string.session_list_item_language_portuguese_content_description);
        break;
    default:
        languageName = context.getString(R.string.session_list_item_language_content_description, languageName);
}
return languageName;
```

Reply:
Converting the `Session` class has to be done with care. The `equals` and `hashCode` function do not cover all fields. I would not do it as part of this branch.

Reply:
The conversion should be done separately and not here...totally agree...just say we could prepare it here since the portion of code has been already touched...so switch-case would also work.